open_spiel/python/examples/lewis_signaling_dqn.py

```diff
@@ -0,0 +1,255 @@
+flags.DEFINE_integer("eps_decay_steps", 49000,
+                     "Number of episodes to decay epsilon")
+
+# Misc paramters
+flags.DEFINE_integer("num_runs", 10, "Number of repetitions")
+flags.DEFINE_integer("log_interval", 100,
+                     "Number of episodes between each logging")
+flags.DEFINE_bool("plot", False, "Set to plot the graphs")
+
+
+def main(_):
+    game = "lewis_signaling"
+    num_players = 2
+
+    num_states = FLAGS.num_states
+    num_messages = FLAGS.num_messages
+    if FLAGS.payoffs == "random":
+        payoffs = np.random.random((num_states, num_states))
+        payoffs_str = ",".join([str(x) for x in payoffs.flatten()])
+    elif FLAGS.payoffs == "climbing":
+        payoffs = np.array([[11, -30, 0], [-30, 7, 6], [0, 0, 5]]) / 30
+        payoffs_str = ",".join([str(x) for x in payoffs.flatten()])
+    else:
+        payoffs_str = FLAGS.payoffs
+        try:
+            payoffs_list = [float(x) for x in payoffs_str.split(",")]
+            payoffs = np.array(payoffs_list).reshape((num_states, num_states))
+        except ValueError:
+            raise ValueError(
+                'There should be {} (states * actions) elements in payoff. Found {} elements'
+                .format(num_states * num_states, len(payoffs_list)))
+
+    env_configs = {
+        "num_states": num_states,
+        "num_messages": num_messages,
+        "payoffs": payoffs_str
+    }
+
+    env = rl_environment.Environment(game, **env_configs)
+    state_size = env.observation_spec()["info_state"][0]
+    num_actions = env.action_spec()["num_actions"]
+    replay_buffer_capacity = int(1e4)
+
+    # Results to store
+    num_runs = FLAGS.num_runs
+    training_episodes = FLAGS.num_episodes
+    log_interval = FLAGS.log_interval
+    rewards = np.zeros((num_runs, training_episodes // log_interval))
+    opts = np.zeros((num_runs, training_episodes // log_interval))
+    converge_point = np.zeros((num_states, num_states))
+    percent_opt = 0
+
+    # Repeat the experiment num_runs times
+    for i in range(num_runs):
+        with tf.Session() as sess:
+            agents = [
+                dqn.DQN(sess,
+                        player_id=idx,
+                        state_representation_size=state_size,
+                        num_actions=num_actions,
+                        learning_rate=FLAGS.step_size,
+                        replay_buffer_capacity=replay_buffer_capacity,
+                        epsilon_start=FLAGS.eps_init,
+                        epsilon_end=FLAGS.eps_final,
+                        epsilon_decay_duration=FLAGS.eps_decay_steps * 2)
+                for idx in range(num_players)
+            ]
+
+            # 1. Train the agents
+            for cur_episode in range(training_episodes):
+                time_step = env.reset()
+                # Find cur_state for logging. See lewis_signaling.cc for info_state details
+                cur_state = time_step.observations["info_state"][0][3:].index(1)
```

Comment:
how about adding current state to time_step.observation["state"]?

Reply:
I would have to change how rl_environment works. Currently it only stores "info_state", "legal_actions", "current_player" ([rl_environment.py](https://github.com/deepmind/open_spiel/blob/a651ce7fbea858b0be2db98afabb49f7c830d904/open_spiel/python/rl_environment.py#L219)). So, I thought it would be easier to just parse the info_state.